Standards/CCSMB-12.md

```diff
@@ -0,0 +1,90 @@
+# *CCSMB 12:* 3D JSON Array Format
+
+*Author: Dimaguy*
+
+*Version: 1.0.0*
+
+*Last revised: 2024-04-02*
+
+The words MUST, MUST NOT, SHOULD, SHOULD NOT, and MAY are defined in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
+This RFC introduces the 3D JSON Array Format, used for printing 3D Blocks in ComputerCraft using the mod [sc-peripherals](https://github.com/SwitchCraftCC/sc-peripherals)
+
+## Quick information
+
+| Information |                           |
+| ----------- | ------------------------- |
+| Version     | 1.0.0                     |
+| Type        | Model Array file format   |
+| MIME        | `model/3dja`              |
+| Extensions  | `.3dja`                   |
+
+## Technical details
+
+The file MUST be composed of a single JSON object ([RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259), "The JavaScript Object Notation (JSON) Data Interchange Format") with the following ruleset:
+- The writer SHOULD specify global non-default (according to 3dj) common settings for the printed 3D blocks in the global namespace of the JSON object.
+- The writer MAY define a dictionary of presets (which are tint/texture combinations), if this isn't done every block MUST use their own texture and tint for each shape.
+- The writer MUST define an array of block models, following the specifications of the [3dj](https://docs.sc3.io/features/sc-peripherals.html#_3dj-format) format with the addition of a coordinates array (if the user foregoes presets each block model object should be compatible with a 3dj file, the coordinates field SHOULD be disregarded by 3dj-only readers).
+- Each block model MAY override a global setting by redefining it on it's context.
+- Size is 1-based indexed, ordered by X, Y and Z each being unsigned integers. It represents number of blocks in each axis
+- Coordinates are 0-based indexed, ordered by x, y and z each being signed integers. It represents position of blocks in each axis
+- The reader MAY provide builtin presets, which MUST be overriden if the namespace coincides with one in the file.
+- The reader MAY provide means of swapping in external presets profiles for recoloring or retexturing, which MUST override part or the entirety of the namespace in file.
+- The reader SHOULD facilitate the printing of a single block of the entire set by providing facilities to print a coordinate, a label of a block or it's index in the file.
+- The reader SHOULD reject or warn about files whose coordinates don't match boundaries in set in size, if one is set.
```

Comment:
Would allowing a file with out-of-bound coordinates obstruct using the file? If so, consider changing this to "MUST reject [...]".

Reply:
Theoretically it wouldn't obstruct using the file for strictly printing, but it wouldn't work so well for a program that wants to build a structure with these instructions hence a minimal reader only makes use of warning about it.

Reply:
Clarified this in the latest commit.